File: lib/src/animate/animation_builder.dart
library angular2.src.animate.animation_builder;

import "package:angular2/src/core/di.dart" show Injectable;
import "css_animation_builder.dart" show CssAnimationBuilder;
import "browser_details.dart" show BrowserDetails;

@Injectable()
class AnimationBuilder {
  BrowserDetails browserDetails;
  /**
   * Used for DI
   * @param browserDetails
   */
  AnimationBuilder(this.browserDetails) {}
  /**
   * Creates a new CSS Animation
   * @returns {CssAnimationBuilder}
   */
  CssAnimationBuilder css() {
    return new CssAnimationBuilder(this.browserDetails);
  }
}

File: modules/@angular/platform-browser/src/animate/animation_builder.ts
import {Injectable} from 'angular2/src/core/di';

import {CssAnimationBuilder} from './css_animation_builder';
import {BrowserDetails} from './browser_details';

@Injectable()
export class AnimationBuilder {
  /**
   * Used for DI
   * @param browserDetails
   */
  constructor(public browserDetails: BrowserDetails) {}

  /**
   * Creates a new CSS Animation
   * @returns {CssAnimationBuilder}
   */
  css(): CssAnimationBuilder { return new CssAnimationBuilder(this.browserDetails); }
}
